src/rdb/rdb_tx.c

```diff
@@ -1035,15 +1072,16 @@ rdb_tx_query_key_max(struct rdb_tx *tx, const rdb_path_t *kvs, d_iov_t *key_out)
 {
 	struct rdb     *db = tx->dt_db;
 	struct rdb_kvs *s;
+	uint64_t	i;
 	int		rc;
 
-	rc = rdb_tx_query_pre(tx, kvs, &s);
+	rc = rdb_tx_query_pre(tx, kvs, &s, &i);
 	if (rc != 0)
 		return rc;
-	rc = rdb_lc_query_key_max(db->d_lc, db->d_applied, s->de_object, key_out);
+	rc = rdb_lc_query_key_max(db->d_lc, i, s->de_object, key_out);
 	if (rc != 0) {
-		D_ERROR(DF_DB": rdb_lc_query_key_max d_applied="DF_U64", rdb_oid="DF_U64"\n",
-			DP_DB(db), db->d_applied, s->de_object);
+		D_ERROR(DF_DB": rdb_lc_query_key_max index="DF_U64", rdb_oid="DF_U64"\n",
```

Comment:
question only (no recommendation to change unless you think it's useful) would it be helpful to know both index and d_applied values in this scenario, to know if the query tried to look up beyond the applied index?

Reply:
That kind of defeats the locking rule some a little extent. But we are not really parallel and this is just logging, so I've added d_applied.